pkg/kubectl/cmd/auth/ican.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/spf13/cobra"
+
+	authorizationapi "k8s.io/api/authorization/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	authorizationclient "k8s.io/client-go/kubernetes/typed/authorization/v1"
+	rbacv1helpers "k8s.io/kubernetes/pkg/apis/rbac/v1"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+	"k8s.io/kubernetes/pkg/printers"
+	util "k8s.io/kubernetes/pkg/printers/internalversion"
+	"k8s.io/kubernetes/pkg/registry/rbac/validation"
+)
+
+type ICanOptions struct {
+	Namespace       string
+	NoHeader        bool
+	ResourceOnly    bool
+	NonResourceOnly bool
+	Quiet           bool
+	SelfSARClient   authorizationclient.SelfSubjectRulesReviewsGetter
+
+	genericclioptions.IOStreams
+}
+
+var (
+	iCanLong = templates.LongDesc(`
+		Check what a user can do in a specific namespace.`)
+
+	iCanExample = templates.Examples(`
+		# Check to see what I can do in current namespace
+		kubectl auth i-can
+
+		# Check to see what I can do in current namespace without header information
+		kubectl auth i-can --no-headers
+
+		# Check to see what I can do with resources in current namespace
+		kubectl auth i-can --only-resources
+
+		# Check to see what I can do with nonResourceURLs in namespace default
+		kubectl auth i-can --only-non-resource-urls --namespace=default`)
+)
+
+func NewCmdICan(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &ICanOptions{
```

Comment:
Was there any effort to include this functionality as part of `kubectl auth can-i --list` maybe? Honestly I was confused what the command does until I read its full description. Wdyt?